### PR TITLE
OpenTelemetry: add SpanLink support

### LINKED ISSUE #684: OpenTelemetry: add SpanLink support
For more details, see the spec: https://opentelemetry.io/docs/instrumentation/java/manual/#create-spans-with-links

### PR DESCRIPTION
**Pull request by IvanFinochenko**

### CODE REVIEW

**Comment by IvanFinochenko** [comment]
ISSUE https://github.com/zio/zio-telemetry/issues/684

File: opentelemetry/src/main/scala/zio/telemetry/opentelemetry/tracing/Tracing.scala [hunk 1]
```diff
@@ -747,15 +766,22 @@ object Tracing {
...
+                           .addLinks(links)
                            .startSpan()
                        )
             } yield (endSpan(span), parentCtx.`with`(span))
 
+          implicit class SpanBuilderOps(spanBuilder: SpanBuilder) {
+            def addLinks(links: Seq[SpanContext]): SpanBuilder =
+              links.foldLeft(spanBuilder) { case (builder, link) => builder.addLink(link) }
+          }
+
```
Comment (grouzen) [inline opentelemetry/src/main/scala/zio/telemetry/opentelemetry/tracing/Tracing.scala]: It makes sense to hide it from the end-user by adding a `private` modifier.

**Comment by grouzen** [comment]
Hey @IvanFinochenko! Great stuff! Thanks for the contribution!
There is only one minor comment to address.

File: opentelemetry/src/main/scala/zio/telemetry/opentelemetry/tracing/Tracing.scala (hunk 1 above)
Comment (IvanFinochenko) [reply opentelemetry/src/main/scala/zio/telemetry/opentelemetry/tracing/Tracing.scala]: done

**Review by grouzen — APPROVED** [review]

**Pull request merged**